internal/storage/storage.go

```diff
@@ -0,0 +1,80 @@
+package storage
+
+import (
+	"context"
+	"encoding/base64"
+	"github.com/jakub-dzon/k4e-operator/api/v1alpha1"
+	"github.com/jakub-dzon/k4e-operator/models"
+	obv1 "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+)
+
+type Claimer struct {
+	client client.Client
+}
+
+func NewClaimer(client client.Client) *Claimer {
+	return &Claimer{client: client}
+}
+
+func (c *Claimer) GetClaim(ctx context.Context, name string, namespace string) (*obv1.ObjectBucketClaim, error) {
+	obc := obv1.ObjectBucketClaim{}
+	err := c.client.Get(ctx, client.ObjectKey{Namespace: namespace, Name: name}, &obc)
+	return &obc, err
+}
+
+func (c *Claimer) CreateClaim(ctx context.Context, device *v1alpha1.EdgeDevice) (*obv1.ObjectBucketClaim, error) {
+	obc := obv1.ObjectBucketClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      device.Name,
+			Namespace: device.Namespace,
+		},
+		Spec: obv1.ObjectBucketClaimSpec{
+			GenerateBucketName: device.Name,
+			StorageClassName:   "openshift-storage.noobaa.io",
+		},
+	}
+	err := c.client.Create(ctx, &obc)
+	return &obc, err
+}
+
+func (c *Claimer) GetStorageConfiguration(ctx context.Context, device *v1alpha1.EdgeDevice) (*models.S3StorageConfiguration, error) {
+	obc, err := c.GetClaim(ctx, *device.Status.DataOBC, device.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	cm := corev1.ConfigMap{}
+	err = c.client.Get(ctx, client.ObjectKey{Namespace: device.Namespace, Name: obc.Name}, &cm)
+	if err != nil {
+		return nil, err
+	}
+	conf := &models.S3StorageConfiguration{
+		BucketHost: cm.Data["BUCKET_HOST"],
+		BucketName: cm.Data["BUCKET_NAME"],
+	}
+	port, err := strconv.Atoi(cm.Data["BUCKET_PORT"])
+	if err == nil && port > 0 {
+		conf.BucketPort = int32(port)
+	}
+
+	secret := corev1.Secret{}
+	err = c.client.Get(ctx, client.ObjectKey{Namespace: device.Namespace, Name: obc.Name}, &secret)
+	if err != nil {
+		return nil, err
+	}
+	awsAccessKeyID, exist := secret.Data["AWS_ACCESS_KEY_ID"]
+	if !exist {
+		return nil, err
+	}
+	conf.AwsAccessKeyID = base64.StdEncoding.EncodeToString(awsAccessKeyID)
+	awsSecretAccessKey, exist := secret.Data["AWS_SECRET_ACCESS_KEY"]
+	if !exist {
+		return nil, err
+	}
+	conf.AwsSecretAccessKey = base64.StdEncoding.EncodeToString(awsSecretAccessKey)
+	return conf, nil
```

Comment:
the returned object should look like:
```
"storageConf": {
  "aws_access_key_id": "MTU3SmRSVjNyV0xObUhTUmZ6QTg=",
  "aws_secret_access_key": "K1RBdWVmbWZZZk95eFMvRXNrRlVzaTNJUjhqaHJ2bU9kdHNzYThkRA==",
  "bucket_host": "s3.openshift-storage.svc",
  "bucket_name": "07b68c4e-41b0-4cb9-87ef-17-cf93bc4e-385c-4545-be64-ac3f8e07427c",
  "bucket_port": 443
}
```

Reply:
The agent/device needs the `bucket_host`to be a routable fqdn in cluster's domain.
_Accessing a bucket from outside the cluster_ section of https://blog.oddbit.com/post/2021-02-10-object-storage-with-openshift lists a script, that prepares bucket_host that can be used by a device.

Reply:
You're right, I'll update the returned configuration to return:
```
{
  "aws_access_key_id": "enZvYWNWeU9UcUp4d29vRkgxanY=",
  "aws_secret_access_key": "dW5idWdjTHlOeXlVQ21HZEF4aU8zOEwwWU5TZzdrNm8rb01tSmZXNg==",
  "bucket_host": "s3-openshift-storage.apps.test-infra-cluster-f13cd1b3.redhat.com",
  "bucket_name": "07b68c4e-41b0-4cb9-87ef-17-97e6547c-5da4-4e96-b001-88302ceb9c32",
  "bucket_port": 443
}
```

Reply:
With this values, it is possible to connect to the s3 endpoint successfully.
I also added the `AWS_CA_BUNDLE` property to store the required certificate for the SSL endpoint (base64 encoded, as the access-key and the secret-access-key).
```
AWS_SECRET_ACCESS_KEY=unbugcLyNyyUCmGdAxiO38L0YNSg7k6o+oMmJfW6
AWS_ACCESS_KEY_ID=zvoacVyOTqJxwooFH1jv
AWS_CA_BUNDLE=/tmp/s3.crt
BUCKET_PORT=443
BUCKET_NAME=07b68c4e-41b0-4cb9-87ef-17-97e6547c-5da4-4e96-b001-88302ceb9c32
BUCKET_HOST=s3-openshift-storage.apps.test-infra-cluster-f13cd1b3.redhat.com
→ aws s3  --endpoint https://$BUCKET_HOST ls --human-readable --summarize
2021-09-05 15:12:22 07b68c4e-41b0-4cb9-87ef-17-97e6547c-5da4-4e96-b001-88302ceb9c32

Total Objects: 0
   Total Size: 0 Bytes
```